lib/raw_net_capture.rb

```diff
@@ -41,6 +41,14 @@ def received(data)
   def sent(data)
     @raw_sent << data
   end
+
+  def headers_only!
+    separator = "\r\n\r\n"
+    if headers_end_index = @raw_received.string.index(separator)
+      @raw_received.truncate(headers_end_index+separator.length)
+    end
+    !!headers_end_index
```

Comment:
U R SO SMRT
let's do that

On Saturday, January 24, 2015, Victor Kmita <notifications@github.com>
wrote:

> In lib/raw_net_capture.rb
> <https://github.com/zendesk/raw_net_capture/pull/2#discussion_r23497765>:
>
> > @@ -41,6 +41,14 @@ def received(data)
> >    def sent(data)
> >      @raw_sent << data
> >    end
> > +
> > +  def headers_only!
> > +    separator = "\r\n\r\n"
> > +    if headers_end_index = @raw_received.string.index(separator)
> > +      @raw_received.truncate(headers_end_index+separator.length)
> > +    end
> > +    !!headers_end_index
>
> Does it not make more sense to have a method #headers that just returns a
> dup @raw_received.truncate(headers_end_index+separator.length) instead of
> actually mutating the @raw_received?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/raw_net_capture/pull/2/files#r23497765>.
>